src/hammer-vlsi/synthesis/vivado/__init__.py

```diff
@@ -0,0 +1,104 @@
+from typing import Dict, List, Any
+
+import os
+import errno
+import pkg_resources
+from itertools import chain
+
+from jinja2 import Template
+
+from hammer_vlsi import HammerSynthesisTool, HammerToolStep, deepdict
+
+_temp_dir = 'file_templates'
+_synth_tag = 'vivado_synth'
+
+class VivadoSynth(HammerSynthesisTool):
+
+
+    _base_files = ['board.tcl', 'paths.tcl', 'project.tcl']
+    
+    @property
+    def env_vars(self) -> Dict[str, str]:
+        new_dict = deepdict(super().env_vars)
+        return new_dict
+
+    @property
+    def steps(self) -> List[HammerToolStep]:
+        return self.make_steps_from_methods([
+            self.run_synthesis,
+            self.run_place_and_route,
```

Comment:
I think we may want to do these in separate classes so that we can run synthesis and par separately, particularly for a large design. I'd envision putting common stuff into a `VivadoCommon` class and having separate `VivadoSynthesis` and `VivadoPlaceAndRoute` classes.

Reply:
That is fine.  However, for testing purpose, supposed I finished synthesis, how would I make hammer start place and route on command line?

Reply:
Assuming we have the output of synthesis as `syn-output.json`, we can run par as follows:

```
hammer-vlsi synthesis_to_par -p syn-output.json -o par-run.json
hammer-vlsi par -p par-run.json
```

For a small enough design, we can also try running both with `hammer-vlsi syn-par [...]`.

Reply:
Do I need to do something special with the "tools" variable in __init__.py so hammer is smart enough to know it should create VivadoPlaceAndRoute?